cunumeric/array.py

```diff
@@ -2465,6 +2465,58 @@ def diagonal(
                 raise ValueError("Either axis1/axis2 or axes must be supplied")
         return self._diag_helper(offset=offset, axes=axes, extract=extract)
 
+    @add_boilerplate("indices", "values")
+    def put(
+        self, indices: ndarray, values: ndarray, mode: str = "raise"
+    ) -> None:
+        """
+        Set storage-indexed locations to corresponding values.
+
+        See Also
+        --------
+         numpy.put
+
+        Availability
+        --------
+        Multiple GPUs, Multiple CPUs
+
+        """
+
+        if values.size == 0 or indices.size == 0:
+            return
+
+        if mode not in ("raise", "wrap", "clip"):
+            raise ValueError("clipmode not understood")
```

Comment:
This would match numpy:

```suggestion
            raise ValueError(f"clipmode must be one of 'clip', 'raise', or 'wrap' (got {mode})")

")
```